src/Operation/Update.php

```diff
@@ -167,40 +167,42 @@ public function execute(Server $server)
             throw UnsupportedException::collationNotSupported();
         }
 
-        $updateOptions = [
-            'multi' => $this->options['multi'],
-            'upsert' => $this->options['upsert'],
-        ];
-
-        if (isset($this->options['arrayFilters'])) {
-            $updateOptions['arrayFilters'] = $this->options['arrayFilters'];
-        }
-
-        if (isset($this->options['collation'])) {
-            $updateOptions['collation'] = (object) $this->options['collation'];
-        }
-
         $bulkOptions = [];
 
         if (isset($this->options['bypassDocumentValidation']) && \MongoDB\server_supports_feature($server, self::$wireVersionForDocumentLevelValidation)) {
             $bulkOptions['bypassDocumentValidation'] = $this->options['bypassDocumentValidation'];
         }
 
         $bulk = new Bulk($bulkOptions);
-        $bulk->update($this->filter, $this->update, $updateOptions);
+        $bulk->update($this->filter, $this->update, $this->createUpdateOptions());
 
-        $writeResult = $server->executeBulkWrite($this->databaseName . '.' . $this->collectionName, $bulk, $this->createOptions());
+        $writeResult = $server->executeBulkWrite($this->databaseName . '.' . $this->collectionName, $bulk, $this->createExecuteOptions());
 
         return new UpdateResult($writeResult);
     }
 
+    public function getCommandDocument()
+    {
+        $cmd = ['update' => $this->collectionName, 'updates' => [['q' => $this->filter, 'u' => $this->update] + $this->createUpdateOptions()]];
+
+        if (isset($this->options['writeConcern'])) {
+            $cmd['writeConcern'] = $this->options['writeConcern'];
+        }
+
+        if (isset($this->options['bypassDocumentValidation']) && \MongoDB\server_supports_feature($server, self::$wireVersionForDocumentLevelValidation)) {
```

Comment:
Adding another thing to the list of late-breaking discoveries, I think `getCommandDocument()` will also need to take a `Server` instance. Without it, we'll get an undefined variable error on the second half of this condition.

The various FindAndModify operations appear to have the same issue.

I suppose the Explainable interface will need that `use` statement after all :stuck_out_tongue: